packages/TelegramClient-Core.package/TCCMessage.class/instance/senderName.st

```diff
@@ -0,0 +1,8 @@
+accessing
+senderName
+
+	self chat isChannel ifTrue: [
+		^ self chat title
+	] ifFalse: [
+		^ self sender fullName
+	].
```

Comment:
I'm not sure this is the best way to implement this. There are also messages where the chat is the sender in chats that are not a channel. Hence, I would recommend doing one of these:
- adding a separate MessageSender / Sender / ... class which accepts either a chat or a user during construction and handles the different interface internally (title vs fullName) with an instance var or something like that
- adding a MessageSenderUser and MessageSenderChat (actually, that seems like a bad solution :P)
- adding a common interface for chat and user that allows everything you would want to do with a sender (this could be a very elegant solution, but without explicitness of said interface, it would be hard)

I would personally choose the first, what do you think?

Reply:
Can you provide an example for a message where a non-channel chat is the sender?

Reply:
For example when a basic group gets upgraded to a super group

Reply:
We now added a senderType attribute to TCCMessage. senderName returns the right string according to the senderType.